mlir/test/Transforms/test-symbol-dce.mlir

```diff
@@ -98,3 +98,22 @@ module {
   // CHECK: "live.user"() {uses = [@unknown_symbol]} : () -> ()
   "live.user"() {uses = [@unknown_symbol]} : () -> ()
 }
+
+// -----
+
+// Check that we don't DCE nested symbols if they are used even if nested inside
+// an unnamed region.
+// CHECK-LABEL: module attributes {test.nested_unnamed_region}
+module attributes {test.nested_unnamed_region} {
+  "test.one_region_op"() ({
+    "test.symbol_scope"() ({
+      // CHECK: func @nfunction
+      func.func @nfunction() {
```

Comment:
```suggestion
      func.func nested @nfunction() {
```

I think that is one thing you're trying to fix.

However a simpler test right now would be:

```


module {
  "test.one_region_op"() ({
    module {
        func.func nested @nested() {
            return
        }

        func.func @main() {
            return
        }
    }
    "test.finish"() : () -> ()
  }) : () -> ()
}
```

Where @main should not be removed but @nested should be. Can you also add this test? It can be a good "documentation" as well of the expected behavior.

Reply:
Good idea, simplified to it (I think it actually shows it better, so no need to have both).